src/ConstructiveSolidGeometry/Intervals.jl

```diff
@@ -15,4 +15,80 @@
+@inline _in_angular_interval_open(α::Real, α_int::AbstractInterval{T}) where {T} = 0 < mod(α - α_int.left, T(2π)) < α_int.right - α_int.left
+
+_in_angular_interval_union(val::T, α::AbstractInterval, β::AbstractInterval) where {T<:AbstractFloat} =
+    _in_angular_interval_closed(val, α) || _in_angular_interval_closed(val, β)
+
+function _is_edge_of_angular_interval_union(val::T, α::AbstractInterval, β::AbstractInterval) where {T<:AbstractFloat}
+    tol = 10*geom_atol_zero(T)
+    (_in_angular_interval_union(val+tol, α, β) && !_in_angular_interval_union(val-tol, α, β)) ||
+    (_in_angular_interval_union(val-tol, α, β) && !_in_angular_interval_union(val+tol, α, β))
+ end
+
+function union_angular_intervals(α::AbstractInterval{T}, β::AbstractInterval{T}) where {T} #if no intersection will return nothing
+    tol = 10*geom_atol_zero(T)
+    if α == β
+        return α
+    elseif α.left == α.right && _in_angular_interval_closed(α.left, β)
+        return β
+    elseif β.left == β.right && _in_angular_interval_closed(β.left, α)
+        return α
+    elseif α.left ≥ 0 && β.left ≥ 0 && α.right < T(2π) && β.right < T(2π) #less not less or equal to 2pi
+        return isempty(α ∩ β) ? nothing : α ∪ β
+    else
+        edges = 0
+        θ1 = T(0)
+        θ2 = T(0)
+        if _is_edge_of_angular_interval_union(α.left, α, β)
+            θ1 = α.left
+            edges = edges + 1
+        end
+        if _is_edge_of_angular_interval_union(α.right, α, β)
+            edges == 0 ? θ1 = α.right : θ2 = α.right
+            edges = edges + 1
+        end
+        if _is_edge_of_angular_interval_union(β.left, α, β) && mod(β.left, T(2π)) ≠ mod(α.left, T(2π)) && mod(β.left, T(2π)) ≠ mod(α.right, T(2π))
+            if edges == 0
+                θ1 = β.left
+            elseif edges == 1
+                θ2 = β.left
+            end
+            edges = edges + 1
+        end
+        if edges < 3 && _is_edge_of_angular_interval_union(β.right, α, β) && mod(β.right, T(2π)) ≠ mod(α.left, T(2π)) && mod(β.right, T(2π)) ≠ mod(α.right, T(2π))
+            edges == 1 ? θ2 = β.right : nothing
+            edges = edges + 1
+        end
+        if edges == 2
+            θMin, θMax = minmax(mod(θ1, T(2π)), mod(θ2, T(2π)))
+            return _in_angular_interval_union(θMin + tol, α, β) ? θMin..θMax : (θMax-T(2π))..θMin
+        elseif edges == 0
+            return T(0)..T(2π)
+        end
+    end
+end
+
+function is_intersection_an_interval(α::AbstractInterval{T}, β::AbstractInterval{T}) where {T}
+    if (mod(α.right - α.left, T(2π)) == 0 || mod(β.right - β.left, T(2π)) == 0)
+        return true
+    elseif isnothing(union_angular_intervals(α, β))
+        return false
+    elseif (mod(α.left, T(2π)) == mod(β.right, T(2π)) && !_in_angular_interval_open(α.right, β)) ||
+           (mod(α.right, T(2π)) == mod(β.left, T(2π)) && !_in_angular_interval_open(α.left, β))
+        return false
+    else
+        return true
+    end
+end
```

Comment:
I feel like this function could be shorter.
```julia
function example(a,b,c)
    if a 
         return true
    elseif b && c
         return false
    else
         return true
    end
end
```
can, for example, be simplified to `example(a,b,c) = a && !(b && c)`

Reply:
This is more human readable. Also I believe in your example the case of `else return true` is not covered by `a && !(b && c)`

Reply:
True! Then, let's leave it as is